dart/test/http_client/failed_request_client_test.dart

```diff
@@ -0,0 +1,277 @@
+      expect(throwableMechanism.mechanism.type, 'SentryHttpClient');
+      expect(throwableMechanism.throwable, isA<TestException>());
+
+      final request = eventCall.event.request;
+      expect(request, isNotNull);
+      expect(request?.method, 'GET');
+      expect(request?.url, 'https://example.com?');
+      expect(request?.queryString, 'foo=bar');
+      expect(request?.cookies, 'foo=bar');
+      expect(request?.headers, {'Cookie': 'foo=bar'});
+      expect(request?.other.keys.contains('duration'), true);
+      expect(request?.other.keys.contains('content_length'), true);
+    });
+
+    test('exception gets not reported if disabled', () async {
+      final sut = fixture.getSut(
+        client: createThrowingClient(),
+        captureFailedRequests: false,
+      );
+
+      await expectLater(
+        () async => await sut.get(requestUri, headers: {'Cookie': 'foo=bar'}),
+        throwsException,
+      );
+
+      expect(fixture.hub.captureEventCalls.length, 0);
+    });
+
+    test('exception gets reported if bad status code occurs', () async {
+      final sut = fixture.getSut(
+        client: fixture.getClient(statusCode: 404, reason: 'Not Found'),
+        badStatusCodes: [SentryStatusCode(404)],
+      );
+
+      await sut.get(requestUri, headers: {'Cookie': 'foo=bar'});
+
+      expect(fixture.hub.captureEventCalls.length, 1);
+
+      final eventCall = fixture.hub.captureEventCalls.first;
+      final throwableMechanism = fixture.hub.captureEventCalls.first.event
+          .throwableMechanism as ThrowableMechanism;
+
+      expect(eventCall.stackTrace, isNull);
+      expect(throwableMechanism, isNotNull);
+      expect(throwableMechanism.mechanism.type, 'SentryHttpClient');
+      expect(
+        throwableMechanism.mechanism.description,
+        'This event was captured because the '
+        'request status code was in [404]',
+      );
+
+      final request = eventCall.event.request;
+      expect(request, isNotNull);
+      expect(request?.method, 'GET');
+      expect(request?.url, 'https://example.com?');
+      expect(request?.queryString, 'foo=bar');
+      expect(request?.cookies, 'foo=bar');
+      expect(request?.headers, {'Cookie': 'foo=bar'});
+      expect(request?.other.keys.contains('duration'), true);
+      expect(request?.other.keys.contains('content_length'), true);
+    });
+
+    test(
+        'just one report on status code reporting with failing requests enabled',
+        () async {
+      final sut = fixture.getSut(
+        client: fixture.getClient(statusCode: 404, reason: 'Not Found'),
+        badStatusCodes: [SentryStatusCode(404)],
+        captureFailedRequests: true,
+      );
+
+      await sut.get(requestUri, headers: {'Cookie': 'foo=bar'});
+
+      expect(fixture.hub.captureEventCalls.length, 1);
+    });
+
+    test('close does get called for user defined client', () async {
+      final mockHub = MockHub();
+
+      final mockClient = CloseableMockClient();
```

Comment:
why not using the fixture for this test?

Reply:
It's a different one.